server/docs/getting-started.md

```diff
@@ -142,384 +288,97 @@ async function initWeb3() {
 
-    nextState.playerSecondNumber = action.number;
-  } else if (action.actionType == ActionType.REVEAL) {
-    require(
-    state.stage == Stage.REVEALING,
-    "Cannot apply REVEALING on REVEALING"
-    );
-    nextState.stage = Stage.DONE;
+```typescript
+async function initContract() {
+  let res = await fetch('HighRollerApp.json')
+  let HighRollerAppArtifact = await res.json();
+  let HighRollerApp = TruffleContract(HighRollerAppArtifact);
 
-    nextState.playerFirstNumber = action.playerFirstNumber;
-    nextState.salt = action.salt;
-  } else {
-    revert("Invalid action type");
-  }
-  return abi.encode(nextState);
+  // Set the provider for our contract
+  HighRollerApp.setProvider(web3Provider);
 }
 ```
 
-You should also take a look at `highRoller(randomness)` which takes in the deterministic “random” seed made from the submitted player numbers and outputs the totals for each player.
+----------
 
-```solidity
-function highRoller(bytes32 randomness)
-    public
-    pure
-    returns(uint8 playerFirstTotal, uint8 playerSecondTotal)
-  {
-    (bytes8 hash1, bytes8 hash2,
-    bytes8 hash3, bytes8 hash4) = cutBytes32(randomness);
-    playerFirstTotal = bytes8toDiceRoll(hash1) + bytes8toDiceRoll(hash2);
-    playerSecondTotal = bytes8toDiceRoll(hash3) + bytes8toDiceRoll(hash4);
-  }
-```
+## setupCF()
 
-Back to the javascript.
+Here's what you'll find in `setupCF()`, and there's no need to change any of it. 
 
+```typescript
+async function setupCF() {
+  nodeProvider = new cf.NodeProvider();
+  await nodeProvider.connect();
+}
+```
 
 ----------
 
 
-# HighRoller.js
+## The install() Function
 
+This is where we begin developing **HighRoller**.
 
-----------
+### Counterfactual's Provider and AppFactory Objects
 
+The `install()` function will
+* reset game state
+* instantiate a Counterfactual `Provider`
+* instantiate a Counterfactual `AppFactory` instance.
 
-## encoding, initial state, and resetGameState()
+The `Provider` is our tool for listening and responding to state changes in a virtual channel: it only needs to be given the `nodeProvider`.
 
-Now that we know what HighRoller.sol looks like, we can fill in the encoding for the game:
+The `AppFactory` is how we create new virtual channels, and it needs to know:
+  * the address of the contract which provides the application logic (**HighRoller.sol**)
+  * the encodings for the application state and actions (to maintain the privacy of state and actions)
+  * which `Provider` can listen and propose updates to state.
 
+We'll input the contract address for **HighRoller.sol**, and fill in the details of the `Action` and `AppState` data types for the encodings.
 
 ```typescript
 async function install() {
   resetGameState();
+  
+  const contractAddress = '0x91907355C59BA005843E791c88aAB80b779446c9';
```

Comment:
We mentioned way earlier in the document that we pre-deployed some high roller contract but we need to re-assert this. We should also link to kovan.etherscan.io/address/0x91907355C59BA005843E791c88aAB80b779446c9 and upload the source code for people to verify.